force-app/main/default/classes/RD2_QueueableService.cls

```diff
@@ -285,14 +285,22 @@ public class RD2_QueueableService {
     public static void enqueueOppNamingService(List<Opportunity> updatedOpps) {
         OpportunityNamingService service = new OpportunityNamingService(updatedOpps);
 
-        if (Limits.getQueueableJobs() < Limits.getLimitQueueableJobs() && !Test.isRunningTest()) {
+        if (canQueueNamingService()) {
             System.enqueueJob(service);
 
         } else {
             service.refreshOppNames();
         }
     }
 
+    private static Boolean canQueueNamingService() {
+        RD2_Settings settings = RD2_Settings.getInstance();
```

Comment:
Over here it should be specifically for Gift Entry Mode Only. We only want the job to be synchronous when it's in gift entry mode, not when the created mode is set to `Synchronous`. I think you can introduce another static variable called `giftEntryMode` and in RD2's should run synchronously you can do a || operation to know if the job needs to be done synchronously.

Reply:
Great catch, could set it during `disableRD2AsynchronousProcesses` as long as it's in the same transaction.